src/System.IO.Pipelines.Networking.Tls/Internal/Sspi/Interop.cs

```diff
@@ -0,0 +1,87 @@
+
+        [DllImport(Dll, ExactSpelling = true, SetLastError = true)]
+        internal static extern SecurityStatus EncryptMessage(ref SSPIHandle contextHandle, [In] uint qualityOfProtection, [In, Out] SecurityBufferDescriptor inputOutput, [In] uint sequenceNumber);
+
+        [DllImport(Dll, ExactSpelling = true, SetLastError = true, CharSet = CharSet.Unicode)]
+        internal static extern int QueryContextAttributesW(ref SSPIHandle phContext, [In] ContextAttribute contextFlag, [Out] out ContextStreamSizes sizes);
+
+        [DllImport(Dll, ExactSpelling = true, SetLastError = true, CharSet = CharSet.Unicode)]
+        internal static extern int QueryContextAttributesW(ref SSPIHandle phContext, [In] ContextAttribute contextFlag, [Out] out _SecPkgContext_KeyInfo sizes);
+
+        [DllImport(Dll, ExactSpelling = true, SetLastError = true, CharSet = CharSet.Unicode)]
+        internal static extern int QueryContextAttributesW(ref SSPIHandle phContext, [In] ContextAttribute contextFlag, [Out] out ContextApplicationProtocol protocolInfo);
+
+        [DllImport(Dll, ExactSpelling = true, SetLastError = true, CharSet = CharSet.Unicode)]
+        internal static extern int QueryContextAttributesW(ref SSPIHandle phContext, [In] ContextAttribute contextFlag, [Out] out SessionKey sessionKey);
+
+        [DllImport(Dll, ExactSpelling = true, SetLastError = true, CharSet = CharSet.Unicode)]
+        internal static extern int QueryContextAttributesW(ref SSPIHandle phContext, [In] ContextAttribute contextFlag, [Out] out _SecPkgContext_ConnectionInfo connectionInfo);
+
+        [DllImport(Dll, ExactSpelling = true, SetLastError = true)]
+        internal unsafe static extern SecurityStatus InitializeSecurityContextW(ref SSPIHandle credentialHandle, [In] void* inContextPtr, [In] string targetName, [In] ContextFlags inFlags, [In] int reservedI, [In] Endianness endianness, SecurityBufferDescriptor* inputBuffer, [In] int reservedII, [In, Out] void* newContextPtr, SecurityBufferDescriptor* outputBuffer, [In, Out] ref ContextFlags attributes, out long timeStamp);
+
+        internal const int SP_PROT_TLS1_0_SERVER = 0x00000040;
+        internal const int SP_PROT_TLS1_0_CLIENT = 0x00000080;
+        internal const int SP_PROT_TLS1_0 = (SP_PROT_TLS1_0_SERVER | SP_PROT_TLS1_0_CLIENT);
+
+        internal const int SP_PROT_TLS1_1_SERVER = 0x00000100;
+        internal const int SP_PROT_TLS1_1_CLIENT = 0x00000200;
+        internal const int SP_PROT_TLS1_1 = (SP_PROT_TLS1_1_SERVER | SP_PROT_TLS1_1_CLIENT);
+
+        internal const int SP_PROT_TLS1_2_SERVER = 0x00000400;
+        internal const int SP_PROT_TLS1_2_CLIENT = 0x00000800;
+        internal const int SP_PROT_TLS1_2 = (SP_PROT_TLS1_2_SERVER | SP_PROT_TLS1_2_CLIENT);
+
+        internal const int SP_PROT_NONE = 0;
+
+        // These two constants are not taken from schannel.h. 
+        internal const int ClientProtocolMask = (SP_PROT_TLS1_0_CLIENT | SP_PROT_TLS1_1_CLIENT | SP_PROT_TLS1_2_CLIENT);
```

Comment:
We moved away from choosing defaults in Core, to letting the OS decide for itself, as old versions of .NET would barf when people disabled the protocols it knew about - SSL2, SSL3, TLS1.0. This seems like a step back to hard coding defaults again.